lib/child_process.js

```diff
@@ -16,22 +16,20 @@ const _validateStdio = child_process._validateStdio;
 const setupChannel = child_process.setupChannel;
 const ChildProcess = exports.ChildProcess = child_process.ChildProcess;
 
-exports.fork = function(modulePath /*, args, options*/) {
-
-  // Get options and args arguments.
-  var options, args, execArgv;
-  if (Array.isArray(arguments[1])) {
-    args = arguments[1];
-    options = util._extend({}, arguments[2]);
-  } else if (arguments[1] && typeof arguments[1] !== 'object') {
+exports.fork = function(modulePath, args, options) {
+
+  if (args && typeof args !== 'object')
     throw new TypeError('Incorrect value of args option');
+
+  if (Array.isArray(args)) {
+    options = Object.assign({}, options);
   } else {
+    options = Object.assign({}, args);
     args = [];
-    options = util._extend({}, arguments[1]);
   }
 
   // Prepare arguments for fork:
-  execArgv = options.execArgv || process.execArgv;
+  var execArgv = options.execArgv || process.execArgv;
```

Comment:
Wow, a missing `var`? How did this ever lint?
Also, you could make this `let`, no?